### PR TITLE
Add bwc handling on non-string sort values on `Hit`s

### LINKED ISSUE #1128: [BUG] Sort values in the Hit type are wrongly always deserialized to string
### What is the bug?
Executing a search request that has sorting in it will produce SearchHits results where each Hit will have it's `sort` array field populated. However the `sort` array is always deserialized as string array and if one wants to use this value to repeat the request with `search_after` being set to the returned value of the `sort` this could result in an error from the server if the field type is not actually string. 

### How can one reproduce the bug?
Create an index which has the following mapping
```
"creationDate": {
    "type": "date",
    "format": "date_time"
}
```

Java field definition
```
@Field(name = "creationDate", type = FieldType.Date, format = DateFormat.date_time)
private Date creationDate;

// Setters/Getters
```
Add a few documents that get indexed and then execute a search request followed by another search that utilizes `search_after`. For example
```
NativeQueryBuilder queryBuilder = NativeQuery.builder();
queryBuilder.withQuery(q -> q.matchAll(m -> m));
queryBuilder.withSort(List.of(
    (new SortOptions.Builder()).field(fb -> fb.field("creationDate").order(SortOrder.Asc))
        .build()));
queryBuilder.withPageable(PageRequest.of(0, 1));

List<Object> searchAfter = null;
boolean done = false;
while (!done) {
  queryBuilder.withSearchAfter(searchAfter);

  SearchHits<T> hits = operations.search(queryBuilder.build(), entityClass, getIndexCoordinates());
  List<SearchHit<T>> searchHits = hits.getSearchHits();

  boolean hasResults = searchHits != null && !searchHits.isEmpty();
  if (hasResults) {
    // This will always return List of strings
    searchAfter = searchHits.getLast().getSortValues();
  }

  if (!hasResults || searchAfter == null || searchAfter.isEmpty() || searchAfter.stream()
      .filter(Objects::nonNull)
      .toList()
      .isEmpty() || (searchHits.size() < 1)) {
    done = true;
  }
}
``` 

When executing the above code the server will return something like
```
{
    // metadata
    "hits": {
        // metadata
        "hits": [
            {
                // _source, _index, _id etc
                "sort": [
                    1720256355885
                ]
            }
        ]
    }
}
```

However during deserialization the client will deserialize the `sort` field to 
```
"sort": [
    "1720256355885"
]
```

Thus repeating the request with search after will produce an error. Something like
```
"root_cause": [
            {
                "type": "parse_exception",
                "reason": "failed to parse date field [1720256355885] with format [date_time]: [failed to parse date field [1720256355885] with format [date_time]]"
            }
        ]
```

### What is the expected behavior?
The returned value types of the `sort` field must be maintained as returned from the server.

### What is your host/environment?
SpringBoot application

### Do you have any additional context?

- The issue seems to be related with the internal Hit generated class and the deserializer associated with the `sort` member - check [here](https://github.com/opensearch-project/opensearch-java/blob/main/java-client/src/main/java/org/opensearch/client/opensearch/core/search/Hit.java#L781)
- Elastic java client seems to have addressed the same issue by changing the type of the `sort` member from `List<String>` to `List<FieldValue>`. Check [this](https://github.com/elastic/elasticsearch-java/commit/9b924bb376419d28738f9ae97d145ba77004115a#diff-2fde5bb5df450c5e85868a8d2c2d53a454caefda6814413f57c011d29fc5729aR512) commit

### PR DESCRIPTION
**Pull request by Xtansia**
### Description
Add bwc handling on non-string sort values on `Hit`s

### Issues Resolved
Fixes #1128 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: java-client/src/main/java/org/opensearch/client/opensearch/core/search/Hit.java [hunk 1]
```diff
@@ -265,8 +267,19 @@ public final Long version() {
...
     public final List<String> sort() {
+        return this.sort.stream().map(FieldValue::_toJsonString).collect(Collectors.toList());
+    }
+
+    /**
+     * API name: {@code sort}
+     *
+     * <p><b>NOTE: In version 3.0.0 of opensearch-java, this method will be renamed to replace {@link #sort()}.</b></p>
+     */
+    public final List<FieldValue> sortVals() {
```
Comment (reta) [inline java-client/src/main/java/org/opensearch/client/opensearch/core/search/Hit.java:282]:
```suggestion
    public final List<FieldValue> sortFields() {
```

Comment (reta) [reply java-client/src/main/java/org/opensearch/client/opensearch/core/search/Hit.java:282]: For 3.x (as you mentioned) we could replace `sort()` directly

Comment (Xtansia) [reply java-client/src/main/java/org/opensearch/client/opensearch/core/search/Hit.java:282]: @reta I can change this naming, however it then won't match that already used to do this fix for `SearchRequest.searchAfter`: 
https://github.com/opensearch-project/opensearch-java/blob/2.x/java-client/src/main/java/org/opensearch/client/opensearch/core/SearchRequest.java#L741-L743

Comment (reta) [reply java-client/src/main/java/org/opensearch/client/opensearch/core/search/Hit.java:282]: @Xtansia ah, I see, sure, let's be consistent, thanks @Xtansia

**Review by reta — APPROVED** [review]

**Pull request merged**